specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2023-04-01-preview/Hunts.json

```diff
@@ -0,0 +1,1001 @@
+          "type": "string"
+        },
+        "status": {
+          "description": "The status of the hunt.",
+          "enum": [
+            "New",
+            "Active",
+            "Closed"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "status"
+          },
+          "default": "New"
+        },
+        "hypothesisStatus": {
+          "description": "The hypothesis status of the hunt.",
+          "enum": [
+            "Unknown",
+            "Invalidated",
+            "Validated"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "hypothesisStatus"
+          },
+          "default": "Unknown"
+        },
+        "attackTactics": {
+          "description": "A list of mitre attack tactics the hunt is associated with",
+          "items": {
+            "$ref": "./common/AlertTypes.json#/definitions/AttackTactic"
+          },
+          "type": "array"
+        },
+        "attackTechniques": {
+          "description": "A list of a mitre attack techniques the hunt is associated with",
+          "items": {
+            "description": "Attack Technique",
+            "type": "string"
+          },
+          "type": "array"
+        },
+        "labels": {
+          "description": "List of labels relevant to this hunt ",
+          "items": {
+            "$ref": "../../../common/2.0/types.json#/definitions/Label"
+          },
+          "type": "array"
+        },
+        "owner": {
+          "$ref": "#/definitions/HuntOwner"
+        }
+      },
+      "required": [
+        "displayName",
+        "description"
+      ],
+      "type": "object"
+    },
+    "HuntRelationList": {
+      "description": "List of all the hunt relations",
+      "properties": {
+        "nextLink": {
+          "description": "URL to fetch the next set of hunt relations.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "value": {
+          "description": "Array of hunt relations",
+          "items": {
+            "$ref": "#/definitions/HuntRelation"
+          },
+          "type": "array"
+        }
+      },
+      "type": "object",
+      "required": [
+        "value"
+      ]
+    },
+    "HuntRelation": {
+      "allOf": [
+        {
+          "$ref": "../../../common/2.0/types.json#/definitions/ResourceWithEtag"
+        }
+      ],
+      "description": "Represents a Hunt Relation in Azure Security Insights.",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/HuntRelationProperties",
+          "description": "Hunt Relation properties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "type": "object"
+    },
+    "HuntRelationProperties": {
+      "description": "Describes hunt relation properties",
+      "properties": {
+        "relatedResourceId": {
```

Comment:
>relatedResourceId [](http://example.com/codeflow?start=9&length=17)

Can you please explain a bit about the scenario here ? When the user creates the huntrelation is he expected to have permissions over this related resoruce id ? How does your service get permissions over this resource ? Do you plan to include linked access checks to ensure that the user also has access to this ?

Reply:
When a hunt relation is created, they should already have access to the related resource id. In certain scenarios, the original resource is duplicated to prevent modifying the original. As of now, this is only a one-way linkage to the related resource, as the resources are stored differently (graph store vs cosmosdb). There may be future plans to address this issue.

Reply:
How do you ensure that they have access to the related resource id ?  Do you do a linked access check : https://armwiki.azurewebsites.net/authorization/RBACLinkedAccessCheck.html

Reply:
From the UX perspective, the user should already have access to the resource through RBAC, and they will be able to make relations PUT call with that resource's id.

From our API's point of view, we are only returning the string, not the resource itself. It is up to the user to make the request to that resource's API to get more information.

Reply:
The issue I am trying to point out is that when the user passes in the relatedResourceId in the PUT call , that user may not have access to that particular resource id from azure RBAC, yet by running this operation they are able to influence it. To avoid this you need to add the linked access check for this resource.

Reply:
Even if the user does not have access to the related resource, they are not able to influence that related resource by creating it here through the PUT request. They still need to make an actual GET request to that API to get the resource.
We are only creating a one-way link to the related resource and we are storing the related resource id string. If the related resource permission's get revoked, or gets deleted, our API does not get notified of this as the relation still persists until the relation gets deleted.

Reply:
SO what do you do with the related resource once it gets created .. I assume there would be more to the operation than just storing  it on your side ? (Note : Feel free to email me to get this resolved a bit quicker)

Reply:
Signing off based on offline conversation. Team agrees to add linked access checks for the scenario.